### PR TITLE
5057 restartpolicy default

### LINKED ISSUE #5057: Document default restart policy in description tag
Neither the docs/ nor swaggerui document what the default `restartPolicy` is.  

`pkg/api/types.go` does state that the default is Always.  

I get why this is the default: replication controllers aren't supposed to work with anything but "Always".  But, that is also a default which may not be obvious to users.  *Why would you want to restart something that exited with success* they may rightly ask.  Of course, changing that default would be a pain since it would break all the existing pod definitions.  Perhaps something to consider for a future API version, once we have a runToCompletionController.

### PR DESCRIPTION
**Pull request by RichieEscarez**
- Updated two help topics to clarify that the default = always
- Updated all the types.go files to include statement of what default is (in descriptions)
- Ran "hack/update-swagger-spec.sh" to build output .json files for swagger UI
  (note, it looks like the script was not run in a long while and this output includes several other changes that i did not deliver)

Fixes #5057

### CODE REVIEW

**Comment by davidopp** [comment]
Great, thanks! Just a few comments:

1) I noticed that update-swagger-spec.sh was never updated to generate
${SWAGGER_ROOT_DIR}/v1.json
Can you add to that script a line at the end that does
curl ${SWAGGER_API_PATH}api/v1 > ${SWAGGER_ROOT_DIR}/v1.json
and then re-run the script and add the new file it generates (v1.json) to your PR?

2) Our standard practice on the team is to squash all the commits into one, so please squash your two commits into one

3) After squashing  (or as part of squashing), please add to the commit message "Fixes #5057" so that merging automatically closes the issue.

**Comment by RichieEscarez** [comment]
Thanks David for all the help on this.

I asked Nikhil because i was concerned and he suggested that I leave the .json output as a separate commit since the build i ran picked up a bunch of other changes that werent mine (ill remember to comment on my intentions next time).

Also to share what i learned (and for future reference), you need the following, in this order, to run to build the new .json files:
 1. Environment setup as defined [here](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/devel/development.md) (where GO and Etcd are installed and in your PATH and k8s is cloned to your GOPATH) 
 2. Run "hack/build-go-sh"
 3. Run "hack/update-swagger-spec.sh"

File: api/swagger-spec/v1.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "swaggerVersion": "",
+  "apiVersion": "",
+  "basePath": "",
+  "resourcePath": ""
+ }
```
Comment (davidopp) [inline api/swagger-spec/v1.json]: Please insert newline at end of file.

**Comment by davidopp** [comment]
Hmm, the v1.json file that got generated is empty. That's weird. Maybe just remove your update-swagger-spec.sh changes, and v1.json, from this PR, and then we could fix update-swagger-spec.sh and add v1.json in a followup PR. Sorry for sending you down that road; I'm not sure what the problem is.

**Comment by davidopp** [comment]
LGTM

**Pull request merged**